1-JavaScript/Week4/homework/challenges/3-password-validation.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * Credit to https://adventofcode.com/ for this exercise
+ *
+ * Each object in the passwordList gives a password policy and then the password.
+ * The times field says the minimal and maximal amount of times the letter should be in the password. So 1-3 means at least 1 time, at most 3 times.
+ * The letter field gives which letter should be counted
+ * The password field gives the password
+ *
+ * In the list 2 passwords are valid, the middle one is not as there is no b in the password.
+ *
+ * We expect the output:
+ *
+ * 'abcde' is VALID, a is present 1 times and should have been present at least 1 and at most 3 times
+ * 'cdefg' is INVALID, b is present 0 times and should have been present at least 1 and at most 3 times
+ * 'ccccccccc' is VALID, c is present 9 times and should have been present at least 2 and at most 9 times
+ */
+
+const passwordList = [
+  { times: '1-3', letter: 'a', password: 'abcde' },
+  { times: '1-3', letter: 'b', password: 'cdefg' },
+  { times: '2-9', letter: 'c', password: 'ccccccccc' },
+];
+
+const characterCounter = (password, letter) => {
+  const passwordArray = password.split('');
+  let count = 0;
+  passwordArray.forEach((element) => {
+    if (element == letter) {
+      count++;
+    }
+  });
+  return count;
+};
+
+const checkPassword = (passwordList) => {
+  let result = '';
+  passwordList.forEach((element) => {
+    const count = characterCounter(element.password, element.letter);
+    const timesArray = element.times.split('-');
```

Comment:
well done!! really nice that you split the logic between two different methods so the code is easier to read and maintain.

I only want to suggest that you con use array destructuring here so it would make the code a bit cleaner.

```suggestion
    const [minStr, maxStr] = element.times.split('-');
    const min = parseInt(minStr, 10);
    const max = parseInt(maxStr, 10);
```

As they are representing a range of numbers, it is very common to use the words min and max.

Reply:
Combine array destructuring with split is so nice.👍